devtools/projects/ng-devtools/src/lib/application-services/settings_store_service.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  effect,
+  EnvironmentInjector,
+  inject,
+  Injectable,
+  runInInjectionContext,
+  signal,
+  WritableSignal,
+} from '@angular/core';
+import {ApplicationOperations} from '../application-operations';
+
+export const SETTINGS_STORE_KEY = 'ng-dt-settings';
+
+@Injectable()
+/** Provides an API for storing and preserving settings values. */
+export class SettingsStoreService {
+  private readonly appOperations = inject(ApplicationOperations);
+  private readonly envInjector = inject(EnvironmentInjector);
+  private data: {[key: string]: unknown} = {};
+
+  private resolveValuesLoaded!: () => void;
+  private readonly valuesLoaded = new Promise<void>((res) => {
+    this.resolveValuesLoaded = res;
+  });
+
+  constructor() {
+    this.loadSettingsValues();
+  }
+
+  /**
+   * Get a settings value from the storage by a provided key, as a writable signal.
+   * If the item doesn't exist, a new one will be created.
+   * Updates to the signal value are automatically stored in the storage.
+   *
+   * @param key Unique key (Prefer format: `<name_of_value/option>@<location>`)
+   * @param initialValue Initial value of the writable signal
+   * @returns A writable signal that represents the value.
+   */
+  get<T>(key: string, initialValue: T): WritableSignal<T> {
+    // Create the signal with its initial value and return it synchronously.
+    const value = signal<T>(initialValue);
+
+    this.valuesLoaded.then(() => {
+      if (key in this.data) {
+        // When the data is loaded and if the key exists, update the signal value.
+        value.set(this.data[key] as T);
+      }
+
+      runInInjectionContext(this.envInjector, () => {
+        effect(() => {
+          // Update the storage when the signal is updated.
+          this.data[key] = value();
+          this.appOperations.setStorageItems({[SETTINGS_STORE_KEY]: this.data});
+        });
+      });
```

Comment:
This is the proper way to run an effect with an injector 
```suggestion
        effect(() => {
          // Update the storage when the signal is updated.
          this.data[key] = value();
          this.appOperations.setStorageItems({[SETTINGS_STORE_KEY]: this.data});
        });
      }, {injector: this.envInjector});
```

Reply:
`runInInjectionContext` has only this signature: 

```typescript
function runInInjectionContext<ReturnT>(injector: Injector, fn: () => ReturnT): ReturnT
```

... or I am missing something?

Reply:
Sorry won't line, it's meant to be the option passed to the effect

Reply:
So, I guess we are good with the current approach (with the excl. of using the current injector instead of the env one)?

Reply:
Oh, okay I got you. I thought that `runInInjectionContext` is still required. We have to update our runtime error since it seems to be a bit outdated.